### ISSUE TITLE
subtract with overflow in new line iterator

### ISSUE THREAD

**Issue by BearOve** [cessen/ropey#76]
It appears there is an issue that was introduced by #70.

The issue is an subtract with overflow in https://github.com/cessen/ropey/blob/1029afac9ff0878cb5a68152745fa855dd996d62/src/iter.rs#L1240

The following code reproduces the issue.

```rust
use ropey::Rope;

fn main() {
    dbg!(Rope::from_str("").lines_at(0).collect::<Vec<_>>());
}
```

**Comment by cessen** [comment]
Thanks so much for catching and reporting this!  I fix is on its way.

**Comment by cessen** [comment]
I'll make a release with the fix later today.

**Comment by BearOve** [comment]
Thank you for very quick response. This was above and beyond expectations 👍

**Comment by cessen** [comment]
No problem!  Also, just published v1.6.0 with the bug fix.
